ts/src/main/scala/ch/epfl/ts/engine/Messages.scala

```diff
@@ -47,7 +48,8 @@ object ExecutedAskOrder {
 abstract class WalletState(val uid: Long)
 
 /* Getter */
-case class GetWalletFunds(override val uid: Long) extends WalletState(uid)
+//actorRef to enable ask pattern (future) and keeping verification of the GetWalletFund sender
+case class GetWalletFunds(override val uid: Long,ref:ActorRef) extends WalletState(uid)
```

Comment:
Actually I don't think this is necessary, could it be replaced by using the `sender` method? http://doc.akka.io/api/akka/2.0/akka/actor/ActorContext.html#sender:ActorRef